File: 1/19/js/RemoveNthNodeFromEndOfList_sol2.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} n
 * @return {ListNode}
 */
var removeNthFromEnd = function(head, n) {
    let countOfAllNodes=0, idx=0, pointer=0;
    let tmp=head;
    while(tmp!==null){
        countOfAllNodes++;
        tmp=tmp["next"];
    }
    idx=countOfAllNodes-n;
    if(pointer==idx){
        if(countOfAllNodes===1) return null;
        else return head["next"];
    }
    let fnres=head;
    while(fnres!==null){
        if(pointer+1===idx){
            let curnextnext=fnres["next"]["next"];
            fnres["next"]=curnextnext;
        }
        fnres=fnres["next"];
        pointer++;
    }
    return head;
};

File: 1/19/cpp/RemoveNthNodeFromEndOfList_sol2.cpp
/*
https://leetcode.com/problems/remove-nth-node-from-end-of-list/description/?envType=daily-question&envId=2024-03-03
*/

/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        int countOfAllNodes=0, idx=0, pointer=0;
        ListNode *tmp=head;
        while(tmp!=nullptr){
            countOfAllNodes++;
            tmp=tmp->next;
        }
        idx=countOfAllNodes-n;
        if(pointer==idx){
            if(countOfAllNodes==1) return nullptr;
            else{
                return head->next;
            }
        }
        ListNode *fnres=head;
        while(fnres!=nullptr){
            if(pointer+1==idx){
                ListNode *curnextnext=fnres->next->next;
                fnres->next=curnextnext;
            }
            fnres=fnres->next;
            pointer++;
        }
        return head;
    }
};